packages/block-editor/src/components/link-control/index.js

```diff
@@ -118,7 +123,8 @@ function LinkControl( {
 			'is-loading': isLoading,
 		} );
 
-		const manualLinkEntryTypes = [ 'url', 'mailto', 'tel', 'internal' ];
+		const directLinkEntryTypes = [ 'url', 'mailto', 'tel', 'internal' ];
```

Comment:
We should extract these types into a `CONSTANT` as it is used across the entire component.

Reply:
> We should extract these types into a `CONSTANT` as it is used across the entire component.

Over time, I've begun to question whether we even need these. I feel like I may have left a similar comment recently elsewhere, but: For what purpose do we assign `type` for manually-entered URLs? It seems like a lot of extra work to assign and accommodate, when it may just be enough to assign nothing, and use the _absence of a type_ as an indicator of a value being manually-entered.

```js
isDirectEntry = ! value.type;
```

Reply:
I [raised an Issue here](https://github.com/WordPress/gutenberg/issues/20051).